### PR TITLE
#160451600 Build out REST Endpoints for Article (CRUD)

### PR DESCRIPTION
**Pull request by mindsworth**
#### What does this PR do?
This PR is for the implementation of CRUD functionality for the back-end. It handles the end-points that creates new article, delete article for a specific user, update article for a specific user, get all articles for a specific user, get a single article for a specific user, and get all articles for all users. The PR also covers the testing for the edge cases.

#### Description of Task to be completed?
- end-points that creates new article
- end-point that updates article for a specific user
- end-point that deletes article for a specific user
- end-point that gets all articles for a specific user
- end-point that gets a single article for a specific user
- end-point that gets all articles for all users
- Wrote test for edge cases

#### How should this be manually tested?
This PR can be tested locally by running the server and using application like postman to make http request to the endpoints.

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#160451600- Users should be able to create articles](https://www.pivotaltracker.com/story/show/160451600)

#### Screenshots (if appropriate)
None

#### Questions:
None

### CODE REVIEW

File: tests/emailController.test.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
-import chai from 'chai';
-import chaiHttp from 'chai-http';
-import app from '..';
+// import chai from 'chai';
```
Comment (KaiserPhemi) [inline tests/emailController.test.js]: #### Critical
Why do we have the entire test file commented out? If it's not needed then delete it but I'll like to know why this is commented out in the first place

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]

**Comment by KaiserPhemi** [comment]
@mindsworth 
Your PR message is not properly formatted. Please fix this.
Also, fix all merge conflicts.

File: server/controllers/ArticleController.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
+import slug from 'slug';
+import uuid from 'uuid';
+
+import { Article } from '../db/models';
+
+const userId = 1;
+
+/**
```
Comment (KaiserPhemi) [inline server/controllers/ArticleController.js:6]: #### Nitpick
Format this properly.

File: server/controllers/ArticleController.js [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+ *
+ * @class ArticleController
+ *
+ * @desc This is a class controller
+ * that handles all operations related to Articles.
+ */
+class ArticleController {
+  /**
+   *@description Add a new Article to the Article list
+   *
```
Comment (KaiserPhemi) [inline server/controllers/ArticleController.js]: #### Nitpick
Remove this blank line. Also remove other  blank lines in comment blocks

File: server/fixtures/auth.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      id: user.id,
+      email: user.email,
+    };
+    const token = jwt.sign(payLoad, 'secret', { expiresIn: '1hr' });
+    res.status(200).json({
+      message: 'Success',
+      token,
+    });
+
+    // res.status(200).json({ status: 'Success', token, });
```
Comment (KaiserPhemi) [inline server/fixtures/auth.js]: #### Nitpick
Remove commented code lines/blocks. If it's not needed, delete it.

**Review by KaiserPhemi — CHANGES_REQUESTED** [review]
- Implement feedback.

**Comment by KaiserPhemi** [comment]
@mindsworth 
Please fix the Code Coverage. It decreased by 8%

**Review by KaiserPhemi — APPROVED** [review]

**Review by Mrshinezee — APPROVED** [review]
LGTM
